docs/jwst/user_documentation/running_pipeline_python.rst

```diff
@@ -529,3 +529,166 @@ individual step parameter must be set when using this method, or else the coded
 defaults will be used, which may be inappropriate for the dataset being processed.
 
 See :ref:`call_examples` for more information.
+
+
+.. _multiprocessing:
+
+Multiprocessing
+===============
+
+Multiprocessing is supported to speed up certain computationally-intensive steps
+in the pipeline, including :ref:`jump_step <jump_step>`,
+:ref:`ramp_fitting_step <ramp_fitting_step>`, and
+:ref:`wfss_contam_step <wfss_contam_step>`. The examples below show how
+multiprocessing can be enabled for these steps, as well as how to set up
+multiprocessing to simultaneously run the entire pipeline on multiple observations.
+
+Python's multiprocessing module explicitly imports and executes a script's
+`__main__` with each and every worker. If `__main__` is not present the behavior is
+undefined. Hence, Python will crash unless the multiprocess code in enclosed in a
+`__main__` block like this:
+
+
+::
+
+    import sys
+
+    def main():
+        [code used in multiprocessing]
+
+    if __name__ = '__main__':
+        sys.exit(main())
+
+
+There are a couple of scenarios to use multiprocessing with the pipeline:
+
+1.  Multiprocessing within a pipeline step. At the moment, the steps that
+support this are the :ref:`jump_step <jump_step>`,
+:ref:`ramp_fitting_step <ramp_fitting_step>`,
+and :ref:`wfss_contam_step <wfss_contam_step>`. To enable multiprocessing the
+optional parameters are `max_cores` for the ``jump`` step, and `maximum_cores`
+for the ``ramp_fitting`` and ``wfss_contam`` steps. These parameters can be
+set to `quarter`, `half`, `all`, or `none`, which is the default value.
+
+The following example turns on a step's multiprocessing option. Notice only
+one of the steps has multiprocessing turned on. We do not recommend to
+simultaneously enable both steps to do multiprocessing, as this may likely
```

Comment:
```suggestion
one of the steps has multiprocessing turned on. We do not recommend
simultaneously enabling both steps to do multiprocessing, as this may likely
```